docs/manual-EN/1.overview/3.design-and-architecture/2.storage-design.md

```diff
@@ -1 +1,136 @@
+
+There may be multiple edge types or multiple edges with the same edge type between two vertices. For example, define an edge type 'transfer', user A may transfer money to user B multiple times, thus a field `rank` is added to distinguish the transfer records between the two. Edge key format shown as Picture Three:
+
+---
+![](https://oscimg.oschina.net/oscnet/3670bf726d129f52f72c1df7fd63f361aea.jpg)
+**Picture Three Edge Key Format**
+
+---
+
+- `Type`: one byte, used to indicate key type, current types are data, index, system, etc.
+- `Part ID`: three bytes, used to indicate the sharding partition. This field can be used to **scan the partition data based on the prefix when re-balancing the partition**
+- `Vertex ID`: four bytes, used to indicate source vertex ID in out-edge, dest vertex ID in in-edge
+- `Edge Type`: four bytes, used to indicate edge type, greater than zero means out-edge, less than zero means in-edge
+- `Rank`: four bytes, used to indicate multiple edges in one edge type. Users can set the filed based on needs, it can store weight such as _transaction time_, _transaction number_ and _weight of a certain order_
+- `Timestamp`: eight bytes, not available to users, used in MVCC in the future
+
+If the value of an edge type is greater than zero, the corresponding edge key format is shown in Picture Four; otherwise if the value less than zero, the corresponding edge key format is shown in Picture Five.
+
+---
+![](https://oscimg.oschina.net/oscnet/2a63e70a5c8bfdbde852c9be33037991e1d.jpg)
+**Picture Four Out-key format**
+
+---
+
+![](https://oscimg.oschina.net/oscnet/e0975916b2689c148cab0cf938d903292f6.jpg)
+**Picture Five In-key format**
+
+---
+
+The properties of vertices and edges are stored in the values of the corresponding kv pairs after encoding. As a strong typed database, nebula will get schema data in the Meta Service before decoding. Besides, when encoding properties, schema version is added to support online schema change.
+
+Nebula shards data through `modulo operation` on vertex ID. All the _out-keys_, _in-keys_ and _tag data_ are placed in the same partition through `modulo operation` on vertex ID. In this way, query efficiency is increased dramatically. For on-line queries, the most common operation is to do BFS expansion from a vertex, thus it's the basic operation to get a vertex's in-edge or out-edge, and such operation's performance determines the traverse's performance. In BFS, there are cases pruning on some properties, nebula guarantees the operation efficiency by putting properties and vertices and edges together. At present, most graph databases verify their efficiency with Graph 500 or Twitter data set, which are, of no eloquence for properties are not included. While most cases use property graphs and BFS needs a lot of pruning as well.
```

Comment:
Breadth-First-Search (BFS)

pruning -> filtering

I made some change to the Chinese version